### PR TITLE
[WIP] API v2

### PR DESCRIPTION
**Pull request by vivangkumar**

### CODE REVIEW

**Comment by vivangkumar** [comment]
[[3] Ensure parameters read from URLs are always correctly decoded (v2)](https://trello.com/c/oexc1Crw/1240-3-ensure-parameters-read-from-urls-are-always-correctly-decoded-v2)

File: chatkit-core/src/main/kotlin/com/pusher/chatkit/users/UserService.kt [hunk 1]
```diff
@@ -24,9 +24,11 @@ internal class UserService(
         val users = userIds.map { id -> getLocalUser(id).orElse { id } }
         val missingUserIds = Result.failuresOf(users).map { it }
         val localUsers = Result.successesOf(users)
+        val missingUserIdsQs = missingUserIds.map { userId -> "id={$userId}"}.joinToString(separator = "&")
```
Comment (hamchapman) [inline chatkit-core/src/main/kotlin/com/pusher/chatkit/users/UserService.kt]: Do we need to escape the `userId` here?

Comment (vivangkumar) [reply chatkit-core/src/main/kotlin/com/pusher/chatkit/users/UserService.kt]: It wasn't being done previously, so I assumed the http library would take care of it..will look into it.

**Comment by vivangkumar** [comment]
Tests will no pass because switching to v2 means subscribing to memberships to populate the `user.rooms` property, instead of depending on the `initial_state` event from the user subscription.

**Comment by mdpye** [comment]
Let's close this for now and then open each other V2 improvement as a PR against this branch.

**Pull request closed without merging**